pkg/scheduler/cache/cache.go

```diff
@@ -288,8 +311,12 @@ func (cache *schedulerCache) processPodResourcesResizeRequest(newPod *v1.Pod) er
 		if podCondition.Type == v1.PodResourcesResizeStatus {
 			actionVer, _ := newPod.ObjectMeta.Annotations[api.AnnotationResizeResourcesActionVer]
 			if podCondition.Message == actionVer {
-				// TODO: If ResizeStatus shows failure, restore previous values
-				//previous, _ :=   pod.ObjectMeta.Annotations[schedulerapi.AnnotationResizeResourcesPrevious]
+				// If ResizeStatus shows failure, restore previous resource values
+				if podCondition.Status == v1.ConditionFalse {
+					if previousResources, ok := newPod.ObjectMeta.Annotations[api.AnnotationResizeResourcesPrevious]; ok {
+						restoreResources(newPod, previousResources)
```

Comment:
restoreResources could return err. this will swallow it without handling it.

Reply:
The error can only happen if the Marshall resulted in corrupted data and unmarshal fails - this would be a bug. The best we can do here is log error (done in the restore function). We need to take the delete and update-done actions in order to not get into a loop.